### PR TITLE
LKE-5305 fix CSV import fail due to nodes hidden by schema

### PR DESCRIPTION
**Pull request by Diachk**
https://linkurious.atlassian.net/browse/LKE-5305

### CODE REVIEW

File: src/backend/graphItemService.ts [hunk 1]
```diff
@@ -307,4 +317,30 @@ export class GraphItemService {
...
+  /**
+   * Ensure that the current user has the correct privileges to import nodes and create a report of the nodes imported.
+   *
+   * @param sourceKey
+   * @param rc
+   */
+  private static async assertHasReportNodeAccess(rc: RestClient, sourceKey: string): Promise<void> {
+    // The CSV_PLUGIN node will not be returned by LKE if:
+      // 1. As an admin, you are in strict schema and the category is not declared
+      // 2. As a custom group user, the category is not declared in the schema
```
Comment (aminerhanemi) [inline src/backend/graphItemService.ts:330]: Is that mean all user of custom group have to create the category in the schema before using the csv plugin ?

File: src/backend/graphItemService.ts [hunk 2]
```diff
@@ -307,4 +317,30 @@ export class GraphItemService {
...
+      // 3. As a standard user (Custom group), the category is declared but not available
```
Comment (aminerhanemi) [inline src/backend/graphItemService.ts:331]: the CSV_PLUGIN category is created during the bootstrap and deleted after. it shouldn't be showed in the schema

File: src/backend/graphItemService.ts [hunk 3]
```diff
@@ -307,4 +317,30 @@ export class GraphItemService {
...
     };
   }
+
+  /**
+   * Ensure that the current user has the correct privileges to import nodes and create a report of the nodes imported.
+   *
+   * @param sourceKey
+   * @param rc
+   */
+  private static async assertHasReportNodeAccess(rc: RestClient, sourceKey: string): Promise<void> {
```
Comment (aminerhanemi) [inline src/backend/graphItemService.ts:327]: maybe changing the name of the method to assertCanReadReportNode

**Review by aminerhanemi — CHANGES_REQUESTED** [review]

File: src/backend/graphItemService.ts (hunk 1 above)
Comment (Diachk) [reply src/backend/graphItemService.ts:330]: Yes, if they have the right to edit the schema

File: src/backend/graphItemService.ts (hunk 2 above)
Comment (Diachk) [reply src/backend/graphItemService.ts:331]: Here I mean that someone intentionally created the category and marked it as hidden

File: src/backend/graphItemService.ts (hunk 3 above)
Comment (Diachk) [reply src/backend/graphItemService.ts]: done

File: src/backend/graphItemService.ts (hunk 1 above)
Comment (Diachk) [reply src/backend/graphItemService.ts:330]: To make it simple we can recommend that you are logged in as an admin and Linkurious is not in strict mode, wdyt ?

**Review by aminerhanemi — APPROVED** [review]

**Pull request merged**